Add production note to all instance/project cmds

raddevon opened the pull request:
Closes #6088

fmoor commented on docs/cli/edgedb_project/edgedb_project_info.rst:
```diff
@@ -18,6 +18,11 @@ Description
 
+.. note::
+
+    The ``edgedb project info`` command is not intended for use with production
```
If you do link your project to a remote instance, this one is supported.
```bash
$ edgedb project info
┌───────────────┬───────────────────────────────────────┐
│ Instance name │ do-test                               │
│ Project root  │ /home/fmoor/src/sample-remote-project │
└───────────────┴───────────────────────────────────────┘
```

raddevon replied:
Removed this one too.

fmoor commented on docs/cli/edgedb_project/edgedb_project_init.rst:
```diff
@@ -21,6 +21,11 @@ to convert an existing directory to a project directory, connecting the
 
+.. note::
+
+    The ``edgedb project init`` command is not intended for use with production
```
You _can_ use this with a remote instance if it is linked already. I'm not sure there are any use cases where we recommend doing this though.
```bash
$ edgedb project init --server-instance do-test
No `edgedb.toml` found in `/home/fmoor/src/sample-remote-project` or above
Do you want to initialize a new project? [Y/n]
> Y
Specify the name of EdgeDB instance to use with this project [default: do-test]:
> do-test
Do you want to use existing instance "do-test" for the project? [y/n]
> y
Applying migrations...
Everything is up to date. Revision initial
Project linked
To connect to do-test, run `edgedb`
```

@1st1 should we just say this is "not intended for use with remote instances"?

raddevon replied:
Elvis says this was the intent for adding the ability to link project to instance, so I've removed the note here. Removing the note as well from `unlink` and `info` which are both applicable once a self-hosted production instance is linked, that leaves only `project upgrade` that this note is applicable to under the `project` commands, so I also removed the top-level note (at `docs/cli/edgedb_project/index.rst`) since it only seems to be the case for one command.

fmoor commented on docs/cli/edgedb_project/index.rst:
```diff
@@ -10,6 +10,11 @@ directory gets associated with a specific EdgeDB instance and thus
 
+.. note::
+
+    The ``edgedb instance`` command group is not intended to manage production
+    instances.
```
```suggestion
    The ``edgedb project`` command group is not intended to manage remote
    instances. See individual commands for more details.
```

fmoor commented on docs/cli/edgedb_instance/edgedb_instance_reset_password.rst:
```diff
@@ -18,6 +18,11 @@ Description
 
+.. note::
+
+    The ``edgedb instance reset-password`` command is not intended for use with
```
Actually, maybe this uses the `--admin` flag to connect. It could still be used with "remote" instances but you can't do it remotely. You would have to ssh into the server running the instance. This is speculation on my part at the moment.

raddevon replied:
I've left this one as-is for now.

raddevon commented on docs/cli/edgedb_instance/edgedb_instance_reset_password.rst:
```diff
@@ -18,6 +18,11 @@ Description
+.. note::
+
+    The ``edgedb instance reset-password`` command is not intended for use with
+    production instances.
```
What's the best way to refer to the instances we want to warn about? Maybe "self-hosted production instances" to draw a clear distinction between these and Cloud instances?

raddevon replied:
I've circled back and changed all the notes to read "…not intended for use with self-hosted production instances." I think that should be good, but let me know if you have something better.

raddevon commented on docs/cli/edgedb_instance/edgedb_instance_unlink.rst:
```diff
@@ -18,6 +18,11 @@ Description
 
+.. note::
+
+    The ``edgedb instance unlink`` command is not intended for use with
```
Good catch. I think I went on auto-pilot. 😅

raddevon commented on docs/cli/edgedb_instance/edgedb_instance_create.rst:
```diff
@@ -20,6 +20,11 @@ instance and creating a corresponding credentials file in
+.. note::
+
+    The ``edgedb instance create`` command is not intended for use with
+    production instances.
```
Good call. Added a link to deployment guides.

raddevon commented on docs/cli/edgedb_instance/edgedb_instance_status.rst:
```diff
@@ -18,6 +18,11 @@ Description
 
+.. note::
+
+    The ``edgedb instance status`` command is not intended for use with
```
Just removed this note.

raddevon commented:
I think I've addressed all of your comments, @fmoor. Take another look when you have a chance and let me know if you see other problems or if one of my changes isn't quite right. Thanks for the great review!

fmoor commented on docs/cli/edgedb_instance/edgedb_instance_create.rst:
```diff
@@ -20,6 +20,13 @@ instance and creating a corresponding credentials file in
+.. note::
+
+    The ``edgedb instance create`` command is not intended for use with
+    self-hosted production instances. You can follow one of our
```
This should also not be used for self-hosted staging or test instances. Maybe just drop `production`? Same for all the self-hosted references.

raddevon replied:
Done.

fmoor reviewed (approved):
Looks good!

raddevon commented:
I really appreciate your help on this, @fmoor. 🙏
